packages/openapi-generator/src/parser/schema.ts

```diff
@@ -59,13 +51,28 @@ export function parseSchema(
   }
 
   if (schema.allOf?.length) {
+    if (schema.properties) {
+      logger.info(
+        'Detected schema with allOf and properties in the same level. This was refactored to a schema with allOf only, also containing all the properties from the top level.'
+      );
+      schema.allOf.push({ properties: schema.properties });
+      delete schema.properties;
+    }
     return parseXOfSchema(schema, refs, 'allOf', options);
   }
 
   if (schema.anyOf?.length) {
     return parseXOfSchema(schema, refs, 'anyOf', options);
   }
 
+  if (
```

Comment:
[nth] Please consider adding a comment, so that other developers are aware that the order matters here